File: testprojects/python_main/pipelines/PERF_PY_PIP/code/com/main1/pythondepmanagement_1/graph/PERF_JOIN.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from com.main1.pythondepmanagement_1.config.ConfigStore import *
from com.main1.pythondepmanagement_1.udfs.UDFs import *

def PERF_JOIN(spark: SparkSession, in0: DataFrame, in1: DataFrame, ) -> DataFrame:
    return in0\
        .alias("in0")\
        .join(in1.alias("in1"), (col("in0.c_long") == col("in1.c_long")), "inner")\
        .where(
          (
            ((col("in0.c_int") > lit(-10)) & (col("in1.c_decimal") > lit(-100)))
            & ~ col("in0.c_string").like("%asdasdrewasdasd%")
          )
        )\
        .select(col("in0.cmls_3ds_authntn_mthd").alias("first_name"), col("in1.c_short").alias("c_short"), col("in0.c_int").alias("c_int"), col("in0.c_decimal").alias("c_decimal"))

File: testprojects/scala_main/pipelines/PERF_SC_PIP/code/src/main/scala/org/main/scla_dep_mgmt_change/graph/PERF_JOIN.scala
package org.main.scla_dep_mgmt_change.graph

import io.prophecy.libs._
import org.main.scla_dep_mgmt_change.config.ConfigStore._
import org.main.scla_dep_mgmt_change.config.Context
import org.main.scla_dep_mgmt_change.udfs.UDFs._
import org.main.scla_dep_mgmt_change.udfs._
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object PERF_JOIN {

  def apply(context: Context, in0: DataFrame, in1: DataFrame): DataFrame =
    in0
      .as("in0")
      .join(in1.as("in1"), col("in0.c_short") === col("in1.c_short"), "inner")
      .where(
        !col("in0.c_string")
          .like("%hello sir this is simple where clause%")
          .and(col("in1.c_int") > lit(-1))
      )
      .select(col("in0.first_name").as("first_name"),
              col("in0.c_short").as("c_short")
      )

}
